src/tools/pkcs11-tool.c

```diff
@@ -1233,6 +1254,53 @@ int main(int argc, char * argv[])
 	if (action_count == 0)
 		util_print_usage_and_die(app_name, options, option_help, NULL);
 
+	if (opt_uri) {
+		/* Check that no interfering options were set */
+		if (opt_module || opt_slot_set || opt_slot_index_set || opt_slot_description ||
+				opt_pin || opt_object_label || opt_object_class_str || opt_object_id_len) {
+			fprintf(stderr, "Invalid combination of arguments when --uri used\n");
```

Comment:
I think it would help to have this error message a bit more verbose explaining what arguments can not be used with the uri, otherwise it is hard to track from this error which of the options is the problematic one.